tests/dune

```diff
@@ -1,3 +1,6 @@
 (tests
- (names test_trace_context)
- (libraries opentelemetry))
+ (names test_trace_context test_get_url)
+ (libraries
+  opentelemetry
```

Comment:
I know it's a bit out of left field, but I think we need 3 distinct test blocks here (one for each set of deps). The reason is that, for the next release of this, opam-ci will test each package separately, so tests need to also bear a `(package opentelemetry)` or `(package opentelemtry-client-ocurl)` etc. with corresponding dependencies.

Sorry :sweat_smile: